wordpress-mobile/WordPress-FluxC-Android
# Persist info on mobile friendly themes

This PR adds support for parsing the mobile-friendly taxonomies each WPCOM theme might belong to. There are 3 groups of themes, each group optimised for a different site use case. Currently each group has 4 themes.

For simplicity, `ThemeModel` just adopts a single `String` field to hold the particular mobile-friendly category slug of the theme. Themes have `null` in that field excepts for the ones that belong to the mobile-friendly taxonomy.

The assumption behind using a simple String instead of a more strict enum/IntDef is that if the categories change in the backend it will only cause the _tests_ to fail but FluxC will work normally, parsing the renamed category and flagging the respective themes with it. In the meantime, a client (like wpandroid) will receive an empty list of themes for the category but hopefully won't crash either.

To test:
* Run the tests or
* Launch the example app, log into WPCOM, tap on "THEMES" and then on "FETCH WP.COM THEMES"
* Notice the log window. There should be a listing of the mobile-friendly themes detected.
